x-pack/test/functional_with_es_ssl/fixtures/plugins/alerts/server/plugin.ts

```diff
@@ -9,16 +9,45 @@ import {
   PluginSetupContract as AlertingSetup,
   AlertType,
 } from '../../../../../../plugins/alerts/server';
+import { PluginSetupContract as FeaturesPluginSetup } from '../../../../../../plugins/features/server';
 
 // this plugin's dependendencies
 export interface AlertingExampleDeps {
   alerts: AlertingSetup;
+  features: FeaturesPluginSetup;
 }
 
 export class AlertingFixturePlugin implements Plugin<void, void, AlertingExampleDeps> {
-  public setup(core: CoreSetup, { alerts }: AlertingExampleDeps) {
+  public setup(core: CoreSetup, { alerts, features }: AlertingExampleDeps) {
     createNoopAlertType(alerts);
     createAlwaysFiringAlertType(alerts);
+    features.registerFeature({
+      id: 'alerting_fixture',
+      name: 'alerting_fixture',
+      app: [],
+      privileges: {
+        all: {
+          alerting: {
+            all: ['test.always-firing', 'test.noop'],
```

Comment:
I wonder if we could/should add the index threshold alert in here; probably will end up requiring a bunch of new tests, so good candidate for follow-on issue.

Reply:
I had a test for that originally and decided to remove it as I wasn't sure we wanted the generic suite to rely on a specific built-in type and creating mock IndexThresholds was a bit more verbose than I felt comfortable with (it detracted from what we were actually trying to test which was generic built-in support).

I was hoping to find a way of creating a mock built-in but that doesn't work because you need the real `alerts` feature to grant those privileges :/